### PR TITLE
Aws support

### PR DESCRIPTION
**Pull request by ahilsend**

### CODE REVIEW

File: pkg/keystore/awsKeystore.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+var (
+	unsealKeysFile = "unseal-keys.json.enc"
+	rootTokenFile  = "root-token.enc"
+)
+
+type AwsKeystore struct {
+	bucketName string
+	kmsKeyId   string
+	s3Service  *s3.S3
```
Comment (zegl) [inline pkg/keystore/awsKeystore.go]: Just wondering: Is it better to use S3 with KMS than to use the AWS Secrets Manager?

File: pkg/keystore/awsKeystore.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+)
+
+var (
+	unsealKeysFile = "unseal-keys.json.enc"
+	rootTokenFile  = "root-token.enc"
+)
+
+type AwsKeystore struct {
+	bucketName string
+	kmsKeyId   string
```
Comment (zegl) [inline pkg/keystore/awsKeystore.go]: Should be `ksmKeyID`.

File: pkg/keystore/awsKeystore.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+	rootTokenData, err := json.Marshal(&initResponse.RootToken)
+	if err != nil {
+		return err
+	}
+	err = keystore.s3Put(rootTokenFile, bytes.NewReader(rootTokenData))
+	if err != nil {
+		return err
+	}
+	log.Printf("Root token written to s3://%s/%s", keystore.bucketName, rootTokenFile)
+}
```
Comment (zegl) [inline pkg/keystore/awsKeystore.go:57]: Missing return.

**Comment by zegl** [comment]
```
~/src/vault-init git(PR-2) golint
main.go:26:2: don't use ALL_CAPS in Go names; use CamelCase
main.go:26:2: exported const PROVIDER_GCP should have comment (or a comment on this block) or be unexported
main.go:27:2: don't use ALL_CAPS in Go names; use CamelCase
main.go:36:2: var kmsKeyId should be kmsKeyID
main.go:55:2: var kmsKeyId should be kmsKeyID
```

File: pkg/keystore/awsKeystore.go (hunk 1 above)
Comment (ahilsend) [reply pkg/keystore/awsKeystore.go]: good point. I'll try it out with the secrets manager

**Comment by ahilsend** [comment]
Refactored to use the secretsmanager. Actually a bit simpler than s3. 

Tested it out, and it successfully initialiazed and unsealed a vault in a private aws instance.

**Comment by ahilsend** [comment]
PTAL

**Review by zegl — APPROVED** [review]
LGTM!

**Pull request merged**